tracer/src/Datadog.Trace/RemoteConfigurationManagement/RcmSubscriptionManager.cs

```diff
@@ -64,13 +68,34 @@ public void Replace(ISubscription oldSubscription, ISubscription newSubscription
     {
         lock (_syncRoot)
         {
-            _subscriptions.Remove(oldSubscription);
+            var newSubscriptions = new List<ISubscription>(_subscriptions.Count);
+
+            bool found = false;
 
-            if (!_subscriptions.Contains(newSubscription))
+            foreach (var subscription in _subscriptions)
             {
-                _subscriptions.Add(newSubscription);
+                var subscriptionToAdd = subscription;
+
+                if (subscriptionToAdd == oldSubscription)
+                {
+                    subscriptionToAdd = newSubscription;
+                }
+
+                if (subscriptionToAdd == newSubscription)
+                {
+                    found = true;
+                }
+
+                newSubscriptions.Add(subscriptionToAdd);
```

Comment:
Maybe it's just me, but this hurt my brain probably more than it should have done. Also, AFAICT, it's not strictly the same as the previous. Maybe this is easier to follow, even though it has a lot more duplication?

```suggestion
                if (subscription == oldSubscription)
                {
                    // swap for the new one
                    newSubscriptions.Add(newSubscription);
                    found = true;
                }
                else if (subscription == newSubscription && !found)
                {
                    // add the new one
                    newSubscriptions.Add(subscription);
                    found = true;
                }
                else
                {
                    // all the others
                    newSubscriptions.Add(subscription);
                }
```

Reply:
Is the new version better? If not I'll go with yours

Reply:
I think it's a bit better, yeah. Meh, either way